### DISCUSSION TITLE
Is it possible to relocate a Spack installation?

### DISCUSSION THREAD

**Discussion by jcwright77** [spack/spack#43307] [Q&A] [answered]
The system I have spack on is phase out the mount point I have spack on. It is possible to relocate a spack installation? It seems that rebuilding modules would work. But what about the paths in spack packages like in package/.spack/* ?

**Comment by tgamblin** [comment, marked as answer]
Spack *does* have relocation logic for:
* buildcaches
* [copying](https://spack.readthedocs.io/en/latest/environments.html#configuration-in-spack-yaml) and relocating binaries into a view with `link_type: copy`

Right now, none of that is implemented for prefix to prefix relocation (though I don't think it would be super hard to make a `spack relocate` command to move an entire install tree.

Relocation doesn't work 100% unless you're relocating to a destination path that's the same or shorter length than the source path (because we can't lengthen certain fields in binaries).  We solve that in CI with `spack config add "config:install_tree:padded_length:128"` (see [the tutorial](https://spack-tutorial.readthedocs.io/en/latest/tutorial_binary_cache.html#setting-up-a-binary-cache-mirror)), but normal installs aren't done that way.

So, if you *are* moving to a shorter path, you *could* create buildcaches out of all your installations and reinstall them in the new location. Otherwise your options are to rebuild or to implement the `spack relocate` feature to move a whole tree, which IMO would be useful.
